lib/src/connection.dart

```diff
@@ -409,11 +416,11 @@ abstract class _PostgreSQLExecutionContextMixin
   int get queueSize => _queue.length;
 
   @override
-  Future<PostgreSQLResult> query(
+  Future<PostgreSQLResult?> query(
     String fmtString, {
-    Map<String, dynamic> substitutionValues,
-    bool allowReuse,
-    int timeoutInSeconds,
+    Map<String, dynamic> substitutionValues = const {},
```

Comment:
I'd rather keep these optional parameters as nullable and set a default value just before we use them.